rs/boundary_node/rate_limits/canister/disclose.rs

```diff
@@ -0,0 +1,125 @@
+use ic_cdk::api::time;
+
+use crate::{
+    access_control::{AccessLevel, ResolveAccessLevel},
+    state::Repository,
+    types::{DiscloseRulesArg, IncidentId, RuleId, Timestamp},
+};
+
+pub trait DisclosesRules {
+    fn disclose_rules(&self, arg: DiscloseRulesArg) -> Result<(), DiscloseRulesError>;
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum DiscloseRulesError {
+    #[error("Operation is not permitted")]
+    Unauthorized,
+    #[error("Incident with ID={0} not found")]
+    IncidentIdNotFound(IncidentId),
+    #[error("Rule with ID={0} not found")]
+    RuleIdNotFound(RuleId),
+    #[error(transparent)]
+    UnexpectedError(#[from] anyhow::Error),
+}
+
+pub struct RulesDiscloser<S, A> {
+    pub state: S,
+    pub access_resolver: A,
+}
+
+impl<S, A> RulesDiscloser<S, A> {
+    pub fn new(state: S, access_resolver: A) -> Self {
+        Self {
+            state,
+            access_resolver,
+        }
+    }
+}
+
+impl<S: Repository, A: ResolveAccessLevel> DisclosesRules for RulesDiscloser<S, A> {
+    fn disclose_rules(&self, arg: DiscloseRulesArg) -> Result<(), DiscloseRulesError> {
+        if self.access_resolver.get_access_level() != AccessLevel::FullAccess {
+            return Err(DiscloseRulesError::Unauthorized);
+        }
+        match arg {
+            DiscloseRulesArg::RuleIds(rule_ids) => {
+                disclose_rules(&self.state, time(), &rule_ids)?;
+            }
+            DiscloseRulesArg::IncidentIds(incident_ids) => {
+                disclose_incidents(&self.state, time(), &incident_ids)?;
+            }
+        }
+
+        Ok(())
```

Comment:
I would make FullAccess the special case
```suggestion
        if self.access_resolver.get_access_level() == AccessLevel::FullAccess {
            match arg {
                DiscloseRulesArg::RuleIds(rule_ids) => {
                    disclose_rules(&self.state, time(), &rule_ids)?;
                }
                DiscloseRulesArg::IncidentIds(incident_ids) => {
                    disclose_incidents(&self.state, time(), &incident_ids)?;
                }
            }
            return Ok(());
        }
        Err(DiscloseRulesError::Unauthorized);
```